instrumentation/opentelemetry-instrumentation-django/src/opentelemetry/instrumentation/django/middleware/otel_middleware.py

```diff
@@ -453,6 +453,15 @@ def process_response(self, request, response):
 
         return response
 
+    def _url_is_disabled(self, request):
+        """
+        Avoid `request.get_host` to bypass Django's `ALLOWED_HOSTS` check
+        """
+        url = "{}://{}{}".format(
+            request.scheme, request._get_raw_host(), request.path
+        )
```

Comment:
Will this break anyone who was relying on the previous behavior of `request.build_absolute_uri("?")`

Reply:
In my experience (not proof of anything, more as an example), host checks are intentional, not an accidental feature of any particular lib, reason why I'm proposing this, in fact.

I can put these behind an environment variable, Django settings or just the current code under that method abstraction so that it can more easily be overridden, without having to copy over the whole code.

Reply:
It's not clear to me why the previous implementation appended a `?` to the URL being checked, but I'm wondering if we should keep it to minimize changes in behavior. What do you think @jourdanrodrigues?